creating_packages/getting_started.rst

```diff
@@ -1,377 +1,221 @@
+    $ mkdir hellopkg && cd hellopkg
+    $ conan new hello/0.1 --template=cmake_lib
+    File saved: conanfile.py
+    File saved: src/CMakeLists.txt
+    File saved: src/hello.cpp
+    File saved: src/hello.h
+    File saved: test_package/conanfile.py
+    File saved: test_package/src/CMakeLists.txt
+    File saved: test_package/src/example.cpp
 
-This will generate the following files:
 
-.. code-block:: text
+On the root folder, there is a *conanfile.py* which is the main recipe file, responsible for
+defining our package and the *src* folder that contains the simple C++ "hello" library with a simple generic
+*CMakeLists.txt* to build it, with nothing specific about Conan in it. Also, there is an optional *test_package*
+folder, which contains an *example* application that links the that will require and link with the created package.
+It is not mandatory, but it is useful to check that our package is correctly created.
 
-    conanfile.py
-    test_package
-      CMakeLists.txt
-      conanfile.py
-      example.cpp
-
-On the root level, there is a *conanfile.py* which is the main recipe file, responsible for
-defining our package. Also, there is a *test_package* folder, which contains a simple example
-consuming project that will require and link with the created package. It is useful to make sure
-that our package is correctly created.
-
-Let's have a look at the root package recipe *conanfile.py*:
+Let's have a look at the package recipe *conanfile.py*:
 
 .. code-block:: python
 
-    from conans import ConanFile, CMake, tools
+    from conans import ConanFile
+    from conan.tools.cmake import CMakeToolchain, CMake
+    from conan.tools.layout import cmake_layout
 
     class HelloConan(ConanFile):
         name = "hello"
         version = "0.1"
-        license = "<Put the package license here>"
-        url = "<Package recipe repository url here, for issues about the package>"
-        description = "<Description of hello here>"
+
+        # Binary configuration
         settings = "os", "compiler", "build_type", "arch"
-        options = {"shared": [True, False]}
-        default_options = {"shared": False}
-        generators = "cmake"
-
-        def source(self):
-            self.run("git clone https://github.com/conan-io/hello.git")
-            # This small hack might be useful to guarantee proper /MT /MD linkage
-            # in MSVC if the packaged project doesn't have variables to set it
-            # properly
-            tools.replace_in_file("hello/CMakeLists.txt", "PROJECT(MyHello)",
-                                  '''PROJECT(MyHello)
-    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-    conan_basic_setup()''')
+        options = {"shared": [True, False], "fPIC": [True, False]}
+        default_options = {"shared": False, "fPIC": True}
+
+        # Sources are located in the same place as this recipe, copy them to the recipe
+        exports_sources = "src/*"
+
+        def config_options(self):
```

Comment:
Too much for starting? Maybe remove the `fPIC` option? and explain the options later?

Reply:
This is the recipe exactly as it is created with ``conan new --template=cmake_lib``, not sure if we should hide it from the docs. I think the real feature is to automate that in Conan, so it doesn't need to be in every recipe, but at the moment, not sure what is the best approach.